cmake/tribits/core/std_tpls/FindTPLMPI.cmake

```diff
@@ -20,7 +20,22 @@ if(WIN32 AND TPL_ENABLE_MPI)
   global_set(TPL_MPI_LIBRARIES ${MPI_LIBRARIES})
 endif()
 
-tribits_tpl_find_include_dirs_and_libraries(MPI)
+tribits_tpl_allow_pre_find_package(MPI MPI_ALLOW_PREFIND)
+if(MPI_ALLOW_PREFIND)
+
+  find_package(MPI)
+
+  if(MPI_C_FOUND AND MPI_CXX_FOUND)
+    tribits_extpkg_create_imported_all_libs_target_and_config_file(
+      MPI
+      INNER_FIND_PACKAGE_NAME MPI
+      IMPORTED_TARGETS_FOR_ALL_LIBS MPI::MPI_C MPI::MPI_CXX)
+  endif()
```

Comment:
Do we need the CXX bindings/part of MPI?  My understanding is that this is something that is not commonly built as part of the MPI installations (for OpenMPI at least).  Likewise, I believe we need the `MPI_Fortran` target e.g. for the Fortran code in SEACAS, but I could be mistaken.

Reply:
There is certainly going to need to be more work and scrutiny if `find_package(MPI)` is going to be made the default way to find and specify MPI for Trilinos.  But for now, this is not used by default so we can deal with the more challenging issues later.  (But eventually, conforming to modern CMake standards means moving to `find_package(MPI)`.)